### PR TITLE
Extend the multicluster scheduling capabilities with placement

### PR DESCRIPTION
**Pull request by haoqing0110**
Ref: https://github.com/open-cluster-management-io/OCM/issues/59
Signed-off-by: haoqing0110 <qhao@redhat.com>

### CODE REVIEW

File: content/en/scenarios/extend-multicluster-scheduling-capabilities.md [hunk 1]
```diff
@@ -0,0 +1,288 @@
+---
+title: Extend the multicluster scheduling capabilities with placement
+weight: 1
+---
+
+Open Cluster Management (OCM) is a community-driven project focused on multicluster and multicloud scenarios for Kubernetes applications. In OCM, the multicluster scheduling capabilities are provided by placement. As we have talked about in the previous article [Using the Open Cluster Management Placement for Multicluster Scheduling](https://cloud.redhat.com/blog/using-the-open-cluster-management-placement-for-multicluster-scheduling), you can use placement to filter clusters by label or claim selector, also placement provides some default prioritizers which can be used to sort and select the most suitable clusters.
```
Comment (qiujian16) [inline content/en/scenarios/extend-multicluster-scheduling-capabilities.md:6]: the firs line introducing ocm seems redundant.

File: content/en/scenarios/extend-multicluster-scheduling-capabilities.md [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+​
+
+One of the default prioritizers is ResourceAllocatableCPU and ResourceAllocatableMemory. They provide the capability to sort clusters based on the allocatable CPU and memory. However, when considering the resource based scheduling, there's a gap that the cluster’s "AllocatableCPU" and "AllocatableMemory" are static values that won’t change even if “the cluster is running out of resources". And in some cases, the prioritizer needs more extra data to calculate the score of the managed cluster. For example, there is a requirement to schedule based on resource monitoring data from the cluster. For this reason, we need a more extensible way to support scheduling based on customized scores.
```
Comment (qiujian16) [inline content/en/scenarios/extend-multicluster-scheduling-capabilities.md:9]: Default prioritizers are

File: content/en/scenarios/extend-multicluster-scheduling-capabilities.md [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+
+## How to implement a customized score provider
+
+The example code is in GitHub repo [resource-usage-collect](https://github.com/JiahaoWei-RH/resource-usage-collect). It provides the score of the cluster's available CPU and available memory, which can reflect the cluster’s real-time resource utilization. It is developed with OCM [addon-framework](https://github.com/open-cluster-management-io/addon-framework) and can be installed as an addon plugin to update customized scores into `AddOnPlacementScore`. (This article won't talk many details about addon-framework, referring to [xxx](https://open-cluster-management.io/concepts/) to learn how to develop an addon.)
```
Comment (qiujian16) [inline content/en/scenarios/extend-multicluster-scheduling-capabilities.md:67]: we should consider put this example into the placement repo IMO

File: content/en/scenarios/extend-multicluster-scheduling-capabilities.md [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+
+  In our example, in addition to recalculate and update the score every 60 seconds, the update will also be triggered when the node or pod resource in the managed cluster changes.
+
+### 3. How to calculate the score
+
+The code to calculate the score is in [pkg/addon/agent/calculate.go](https://github.com/JiahaoWei-RH/resource-usage-collect/blob/main/pkg/addon/agent/calculate.go). A valid score must be in the range -100 to 100, you need to normalize the scores before updating it into `AddOnPlacementScore`. 
+
+When normalizing the score, you might meet the below cases.
+
+- The score provider knows the max and min value of the customized scores.
```
Comment (qiujian16) [inline content/en/scenarios/extend-multicluster-scheduling-capabilities.md:268]: we might need to consider to provide a normalize library?

File: content/en/scenarios/extend-multicluster-scheduling-capabilities.md (hunk 1 above)
Comment (haoqing0110) [reply content/en/scenarios/extend-multicluster-scheduling-capabilities.md]: Removed this line.

File: content/en/scenarios/extend-multicluster-scheduling-capabilities.md (hunk 2 above)
Comment (haoqing0110) [reply content/en/scenarios/extend-multicluster-scheduling-capabilities.md]: Updated.

File: content/en/scenarios/extend-multicluster-scheduling-capabilities.md (hunk 4 above)
Comment (haoqing0110) [reply content/en/scenarios/extend-multicluster-scheduling-capabilities.md:268]: Yes, that's sounds like a good suggestion, let me prepare another PR for that.

File: content/en/scenarios/extend-multicluster-scheduling-capabilities.md (hunk 3 above)
Comment (haoqing0110) [reply content/en/scenarios/extend-multicluster-scheduling-capabilities.md:67]: Sure

**Comment by haoqing0110** [comment]
Follow-up actions: 
https://github.com/open-cluster-management-io/OCM/issues/68
https://github.com/open-cluster-management-io/OCM/issues/69

**Pull request merged**